gensim/models/ldaseqmodel.py

```diff
@@ -0,0 +1,1121 @@
+        self.fwd_mean = numpy.array(numpy.split(numpy.zeros((num_sequences + 1) * num_terms), num_terms))
+        self.fwd_variance = numpy.array(numpy.split(numpy.zeros((num_sequences + 1) * num_terms), num_terms))
+        self.variance = numpy.array(numpy.split(numpy.zeros((num_sequences + 1) * num_terms), num_terms))
+        self.zeta = numpy.zeros(num_sequences)
+
+        # the following are class variables which are to be integrated during Document Influence Model
+        self.m_update_coeff = None
+        self.mean_t = None
+        self.variance_t = None
+        self.influence_sum_lgl = None
+        self.w_phi_l = None
+        self.w_phi_sum = None
+        self.w_phi_l_sq = None
+        self.m_update_coeff_g = None
+
+        # temp_vect 
+        self.temp_vect = None
+
+class Lda_Post(utils.SaveLoad):
+    """
+    Posterior values associated with each set of documents document
+
+    """
+
+    def __init__(self, doc=None, lda=None, max_nterms=None, num_topics=None, gamma=None, lhood=None):
+
+        self.doc = doc
+        self.lda = lda
+        self.gamma = gamma
+        self.lhood = lhood
+
+        if max_nterms is not None and num_topics is not None:
+            self.phi = numpy.resize(numpy.zeros(max_nterms * num_topics), (max_nterms, num_topics))
+            self.log_phi = numpy.resize(numpy.zeros(max_nterms * num_topics), (max_nterms, num_topics))
+
+        # the following are class variables which are to be integrated during Document Influence Model
+
+        self.doc_weight = None
+        self.renormalized_doc_weight = None
+
+def update_zeta(sslm):
+
+    num_terms = sslm.num_terms
+    num_sequences = sslm.num_sequences
+    # making zero and updating
+    sslm.zeta.fill(0)
+    for i in range(0, num_terms):
+        for j in range(0, num_sequences):
+            m = sslm.mean[i][j + 1]
+            v = sslm.variance[i][j + 1]
+            val = numpy.exp(m + v/2)
+            sslm.zeta[j] = sslm.zeta[j] + val  
+    return
+
+def compute_post_variance(word, sslm, chain_variance):
+
+    T = sslm.num_sequences
+    variance = sslm.variance[word] # pick wordth row
+    fwd_variance = sslm.fwd_variance[word] # pick wordth row
+
+    # forward pass. Set initial variance very high
+    fwd_variance[0] = chain_variance * 1000
+    
+    for t in range(1, T + 1):
+        if sslm.obs_variance:
+            w = sslm.obs_variance / (fwd_variance[t - 1] + chain_variance + sslm.obs_variance)
+        else:
+            w = 0
+        fwd_variance[t] = w * (fwd_variance[t-1] + chain_variance)
+
+    # backward pass 
+    variance[T] = fwd_variance[T]
+    for t in range(T - 1, -1, -1):
+        if fwd_variance[t] > 0.0:
+            w = numpy.power((fwd_variance[t] / (fwd_variance[t] + chain_variance)), 2)
+        else:
+            w = 0
+        variance[t] = (w * (variance[t + 1] - chain_variance)) + ((1 - w) * fwd_variance[t])
+
+    sslm.variance[word] = variance
```

Comment:
can it just return the new variance and fwd_variance? Functions with side effects are hard to read. Same for `compute_post_mean` below